File: plantillas/terrible.py
import requests
import random

API_URL = "https://tormenta-codigo-app-terrible.vercel.app/api/podcast"

def fetch_episodes():
    try:
        response = requests.get(API_URL)
        response.raise_for_status()  # will raise an error for 4xx and 5xx responses
        return response.json()["data"]
    except requests.RequestException as e:
        print(f"Error fetching the episodes: {e}")
        return []

def process_episodes(episodes):
    if not episodes:
        return

    # Convert duration to integers and sort episodes by number
    for ep in episodes:
        ep['duration'] = int(ep['duration'])
    episodes.sort(key=lambda x: int(x['number']))

    # Calculate the next episode number
    next_episode_number = int(episodes[-1]['number']) + 1

    # Calculate the total duration
    total_duration = sum(ep['duration'] for ep in episodes)

    # Find the shortest episode
    shortest_episode = min(episodes, key=lambda x: x['duration'])

    # Shuffle episodes and select titles that sum to less than 2 hours
    random.shuffle(episodes)
    two_hour_limit = 2 * 60 * 60
    duration_sum = 0
    selected_titles = []
    for ep in episodes:
        if duration_sum + ep['duration'] <= two_hour_limit:
            duration_sum += ep['duration']
            selected_titles.append(ep['title'])

    # Print results
    print(f"Next episode number: {next_episode_number}")
    print(f"Total duration of all episodes: {total_duration}")
    print(f"Number of the shortest episode: {shortest_episode['number']}")
    print(f"Titles below 2 hours: {selected_titles}")

def main():
    episodes = fetch_episodes()
    process_episodes(episodes)

if __name__ == "__main__":
    main()

File: plantillas/terrible.php
<?php

$apiUrl = "https://tormenta-codigo-app-terrible.vercel.app/api/podcast";

function fetchEpisodes($apiUrl)
{
  $ch = curl_init();
  curl_setopt($ch, CURLOPT_URL, $apiUrl);
  curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
  $result = curl_exec($ch);
  curl_close($ch);

  if (!$result) {
    echo "Error fetching the episodes: ", PHP_EOL;
    return [];
  }

  $data = json_decode($result, true);
  return $data['data'];
}

function processEpisodes($episodes)
{
  if (!$episodes) {
    return;
  }

  // Convert duration to integers and sort episodes by number
  foreach ($episodes as &$ep) {
    $ep['duration'] = (int)$ep['duration'];
  }
  unset($ep);

  usort($episodes, function ($a, $b) {
    return (int)$a['number'] - (int)$b['number'];
  });

  // Calculate the next episode number
  $nextEpisodeNumber = (int)$episodes[count($episodes) - 1]['number'] + 1;

  // Calculate the total duration
  $totalDuration = array_sum(array_column($episodes, 'duration'));

  // Find the shortest episode
  $shortestEpisode = min(array_column($episodes, 'duration'));

  // Shuffle episodes and select titles that sum to less than 2 hours
  shuffle($episodes);
  $twoHourLimit = 2 * 60 * 60;
  $durationSum = 0;
  $selectedTitles = [];
  foreach ($episodes as $ep) {
    if ($durationSum + $ep['duration'] <= $twoHourLimit) {
      $durationSum += $ep['duration'];
      $selectedTitles[] = $ep['title'];
    }
  }

  // Print results
  echo "Next episode number: ", $nextEpisodeNumber, PHP_EOL;
  echo "Total duration of all episodes: ", $totalDuration, PHP_EOL;
  echo "Duration of the shortest episode: ", $shortestEpisode, PHP_EOL;
  echo "Titles below 2 hours: ", implode(', ', $selectedTitles), PHP_EOL;
}

$episodes = fetchEpisodes($apiUrl);
processEpisodes($episodes);
